packages/react-dom/src/client/ToStringValue.js

```diff
@@ -44,14 +45,39 @@ export opaque type TrustedValue: {toString(): string, valueOf(): string} = {
   valueOf(): string,
 };
 
+export function evaluateToStringOrTrustedType(
+  value: mixed,
+): string | TrustedValue {
+  if (
+    value !== null &&
+    typeof value === 'object' &&
+    value.$$typeof === REACT_OPAQUE_OBJECT_TYPE &&
+    typeof value._setId === 'function'
+  ) {
+    value._setId();
+    return '';
+  } else {
+    return toStringOrTrustedType(value);
```

Comment:
The point to collocate these was so that you could use the same type check to quickly determine if this is *not* an object which is the common case. However, by calling this second function, you have to check the type again in the other function. Leading to two checks instead of one, for the common case.

You should inline this here so that you can avoid extra type checks. However, the bigger question is why you need two functions at all.